src/parser.rs

```diff
@@ -269,15 +270,56 @@ impl Library {
                         "virtual-method" => try!(ignore_element(parser)),
                         "doc" => doc = try!(read_text(parser)),
                         "union" => {
-                            use self::Type::*;
-                            if let Union(u) = try!(self.read_union(parser, ns_id, attrs)) {
+                            if let Type::Union(mut u) = try!(self.read_union(
+                                parser,
+                                ns_id,
+                                &attributes,
+                                Some(&class_name),
+                                Some(&c_type)
+                            )) {
+                                let mut field_name = String::new();
+                                for attr in &attributes {
+                                    match attr.name.local_name.as_ref() {
+                                        "name" => {
+                                            field_name = attr.value.clone();
+                                            break;
+                                        }
+                                        _ => {}
+                                    }
+                                }
+
+                                if field_name == "" {
+                                    field_name = format!("u{}", union_count);
```

Comment:
`if field_name.is_empty() {` (2 cases)
Bad that we can't remove `union_count` completely.